massa-execution-worker/src/context.rs

```diff
@@ -475,20 +467,48 @@ impl ExecutionContext {
         // create a seed from the current slot
         let mut data: Vec<u8> = self.slot.to_bytes_key().to_vec();
         // add the index of the created address within this context to the seed
-        data.append(&mut self.created_addr_index.to_be_bytes().to_vec());
+        data.extend(self.created_addr_index.to_be_bytes());
         // add a flag on whether we are in read-only mode or not to the seed
         // this prevents read-only contexts from shadowing existing addresses
         if self.read_only {
             data.push(0u8);
         } else {
             data.push(1u8);
         }
-        // hash the seed to get a unique address
-        let hash = Hash::compute_from(&data);
-        let address = self.address_factory.create(
-            &AddressArgs::SC { hash },
-            FactoryStrategy::At(slot_timestamp),
-        )?;
+        // hash the seed to get an address
+        let mut hash = Hash::compute_from(&data);
+        let mut nonce = 0u64;
+        let mut address;
+
+        // Loop over nonces until we find an address that doesn't exist in the speculative ledger.
+        // Note that this loop is here for robustness, and should not be looping because
+        // even through the SC addresses are predictable, nobody can create them beforehand because:
+        // - their category is "SC" and not "USER" so they can't be derived from a public key
+        // - sending tokens to the target SC address to create it by funding is not allowed because transactions towards SC addresses are not allowed
+        loop {
+            address = self.address_factory.create(
```

Comment:
In the first iteration you don't add the nonce to the hash and so it will be complicated to document and explain that if nonce is 0 we don't add one. Maybe it could be better to just add the nonce at the first iteration to have an unify behaviour ?

Reply:
done